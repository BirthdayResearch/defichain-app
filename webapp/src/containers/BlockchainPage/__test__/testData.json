{
  "getBlockHash": {
    "result": "17d34e560a64717f5126272044bb34b911768a6b551059914432a2276670c8ff",
    "error": null,
    "id": "curltest"
  },
  "getBlock": {
    "result": {
      "hash": "17d34e560a64717f5126272044bb34b911768a6b551059914432a2276670c8ff",
      "confirmations": 3,
      "strippedsize": 214,
      "size": 250,
      "weight": 892,
      "height": 99,
      "version": 536870912,
      "versionHex": "20000000",
      "merkleroot": "60f2d8259dba360c265cdae6fc52b149d9f915e73608386a7e8d75ce63d27338",
      "tx": [
        "60f2d8259dba360c265cdae6fc52b149d9f915e73608386a7e8d75ce63d27338"
      ],
      "time": 1590490857,
      "mediantime": 1590490856,
      "nonce": 2,
      "bits": "207fffff",
      "difficulty": 4.656542373906925e-10,
      "chainwork": "00000000000000000000000000000000000000000000000000000000000000c8",
      "nTx": 1,
      "previousblockhash": "1d81f12687c614d2da57d1200fc2f8f1da27103a571fb95e9c3cb7c85cd43e48",
      "nextblockhash": "7ef9b3ffeaa3b600e4d0c664052e45ed53e8f28686d0f006603bf9cbcd927ed4"
    },
    "error": null,
    "id": "curltest"
  },
  "getBlockCount": { "result": 101, "error": null, "id": "curltest" },
  "handelFetchBlocks": {
    "getBlockCount": { "result": 101, "error": null, "id": "curltest" }
  },
  "getrawtransaction": {
    "result": {
      "in_active_chain": true,
      "txid": "60f2d8259dba360c265cdae6fc52b149d9f915e73608386a7e8d75ce63d27338",
      "hash": "a3a67b2ff2ea4cf86807e1c6eaa26d050297c2699a820b743807000c1bf66351",
      "version": 2,
      "size": 169,
      "vsize": 142,
      "weight": 568,
      "locktime": 0,
      "vin": [{ "coinbase": "01630101", "sequence": 4294967295 }],
      "vout": [
        {
          "value": 50.0,
          "n": 0,
          "scriptPubKey": {
            "asm": "0 0cf3cafa8590686c79d9a98481ee1632f20ce5c4",
            "hex": "00140cf3cafa8590686c79d9a98481ee1632f20ce5c4",
            "reqSigs": 1,
            "type": "witness_v0_keyhash",
            "addresses": ["bcrt1qpneu4759jp5xc7we4xzgrmskxteqeewytnxqm9"]
          }
        },
        {
          "value": 0.0,
          "n": 1,
          "scriptPubKey": {
            "asm": "OP_RETURN aa21a9ede2f61c3f71d1defd3fa999dfa36953755c690689799962b48bebd836974e8cf9",
            "hex": "6a24aa21a9ede2f61c3f71d1defd3fa999dfa36953755c690689799962b48bebd836974e8cf9",
            "type": "nulldata"
          }
        }
      ],
      "hex": "020000000001010000000000000000000000000000000000000000000000000000000000000000ffffffff0401630101ffffffff0200f2052a010000001600140cf3cafa8590686c79d9a98481ee1632f20ce5c40000000000000000266a24aa21a9ede2f61c3f71d1defd3fa999dfa36953755c690689799962b48bebd836974e8cf90120000000000000000000000000000000000000000000000000000000000000000000000000",
      "blockhash": "17d34e560a64717f5126272044bb34b911768a6b551059914432a2276670c8ff",
      "confirmations": 3,
      "time": 1590490857,
      "blocktime": 1590490857
    },
    "error": null,
    "id": "curltest"
  },
  "expected": {
    "handleFetchBlockData": {
      "blockData": {
        "nTxns": 1,
        "height": 99,
        "bits": "207fffff",
        "difficulty": 4.656542373906925e-10,
        "hash": "17d34e560a64717f5126272044bb34b911768a6b551059914432a2276670c8ff",
        "merkleRoot": "60f2d8259dba360c265cdae6fc52b149d9f915e73608386a7e8d75ce63d27338",
        "version": 536870912,
        "nonce": 2
      }
    },
    "handleFetchBlockCount": {
      "blockCount": 101
    },
    "handelFetchBlocks": {
      "blocks": [
        {
          "hash": "17d34e560a64717f5126272044bb34b911768a6b551059914432a2276670c8ff",
          "size": 250,
          "height": 99,
          "version": 536870912,
          "merkleRoot": "60f2d8259dba360c265cdae6fc52b149d9f915e73608386a7e8d75ce63d27338",
          "txnIds": [
            "60f2d8259dba360c265cdae6fc52b149d9f915e73608386a7e8d75ce63d27338"
          ],
          "nonce": 2,
          "bits": "207fffff",
          "difficulty": 4.656542373906925e-10,
          "time": "May 26, 04:30 pm",
          "nTxns": 1
        }
      ],
      "blockCount": 101
    },
    "handelFetchTxns": {
      "txns": [
        {
          "hash": "a3a67b2ff2ea4cf86807e1c6eaa26d050297c2699a820b743807000c1bf66351",
          "time": "May 26, 04:30 pm",
          "tos": [
            {
              "amount": "50",
              "address": "bcrt1qpneu4759jp5xc7we4xzgrmskxteqeewytnxqm9"
            },
            {
              "amount": "0",
              "address": "Unparsed Address"
            }
          ],
          "unit": "DFI"
        }
      ],
      "txnCount": 1
    }
  },
  "fetchBlockDataRequestTestData": {
    "nTxns": 0,
    "difficulty": 0,
    "height": 0,
    "bits": "",
    "version": 0,
    "nonce": 0,
    "hash": "",
    "merkleRoot": ""
  }
}
