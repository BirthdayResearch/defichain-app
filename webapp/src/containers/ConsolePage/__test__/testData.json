{
  "handleDataForQuery": {
    "result": "bcrt1qm8afjayussf0uryqf66ppng0p47l6ml09uj28g",
    "error": null,
    "id": "curltest"
  },
  "referenceComponentProps": {
    "title": "sendtoaddress",
    "highlightText": "sendtoaddress \"address\" amount ( \"comment\" \"comment_to\" subtractfeefromamount replaceable conf_target \"estimate_mode\" )",
    "cmdText": "defi-cli sendtoaddress \"1M72Sfpbz1BPpXFHz9m3CdqATR44Jvaydd\" 0.1 \"donation\" \"seans outpost\""
  },
  "expected": {
    "handleDataForQuery": "bcrt1qm8afjayussf0uryqf66ppng0p47l6ml09uj28g"
  }
}
