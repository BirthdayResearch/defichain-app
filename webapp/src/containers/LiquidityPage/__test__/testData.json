{
  "poolPairList": [
    {
      "apy": "20.00",
      "blockCommissionA": 4.8e-7,
      "blockCommissionB": 3e-8,
      "commission": 0.002,
      "creationHeight": 18145,
      "creationTx": "b2bb09ffe9f9b292f13d23bafa1225ef26d0b9906da7af194c5738b63839b235",
      "idTokenA": "0",
      "idTokenB": "1",
      "key": "3",
      "name": "Default Defi token-",
      "ownerAddress": "76a914e5ed1bfa532e2358e964381a70e0df6d4dd0041988ac",
      "poolID": "3",
      "poolSharePercentage": "0",
      "reserveA": 23496.26010905,
      "reserveA/reserveB": 2.77599046,
      "reserveB": 8464.0997198,
      "reserveB/reserveA": 0.36023178,
      "rewardPct": 0.85,
      "status": true,
      "symbol": "DFI-BTC",
      "tokenA": "DFI",
      "tokenB": "BTC",
      "totalLiquidity": 1726.8492621,
      "totalLiquidityInUSDT": "229875757.61792275",
      "tradeEnabled": true,
      "yearlyPoolReward": "56693844.00000000"
    },
    {
      "apy": "252.31",
      "blockCommissionA": 3e-8,
      "blockCommissionB": 1.4e-7,
      "commission": 0.002,
      "creationHeight": 18145,
      "creationTx": "045c74ee64e2afce750d2dae5958a01023a0dd30105cff211b6fb715a6aa915d",
      "idTokenA": "0",
      "idTokenB": "2",
      "key": "4",
      "name": "Default Defi token-",
      "ownerAddress": "76a914e5ed1bfa532e2358e964381a70e0df6d4dd0041988ac",
      "poolID": "4",
      "poolSharePercentage": "0",
      "reserveA": 246193.63969597,
      "reserveA/reserveB": 100.63655021,
      "reserveB": 2446.36406115,
      "reserveB/reserveA": 0.00993674,
      "rewardPct": 0.1,
      "status": true,
      "symbol": "DFI-ETH",
      "tokenA": "DFI",
      "tokenB": "ETH",
      "totalLiquidity": 23423.75299491,
      "totalLiquidityInUSDT": "2143400.30751132",
      "tradeEnabled": true,
      "yearlyPoolReward": "6669864.00000000"
    },
    {
      "apy": "861.38",
      "blockCommissionA": 1.2e-7,
      "blockCommissionB": 1e-8,
      "commission": 0.002,
      "creationHeight": 29584,
      "creationTx": "72fa270eddf0221113f5e29d0e9568f89e87c433c91aad50b1215f0aad60e7c3",
      "idTokenA": "0",
      "idTokenB": "5",
      "key": "6",
      "name": "Default Defi token-",
      "ownerAddress": "76a914e5ed1bfa532e2358e964381a70e0df6d4dd0041988ac",
      "poolID": "6",
      "poolSharePercentage": "0",
      "reserveA": 193976.12476028,
      "reserveA/reserveB": 4.79992181,
      "reserveB": 40412.35093661,
      "reserveB/reserveA": 0.20833672,
      "rewardPct": 0.05,
      "status": true,
      "symbol": "DFI-USDT",
      "tokenA": "DFI",
      "tokenB": "USDT",
      "totalLiquidity": 79409.57427086,
      "totalLiquidityInUSDT": "313913.47365533",
      "tradeEnabled": true,
      "yearlyPoolReward": "3334932.00000000"
    },
    {
      "apy": "0.00",
      "blockCommissionA": 0,
      "blockCommissionB": 2e-7,
      "commission": 0.002,
      "creationHeight": 87550,
      "creationTx": "c38593974982ce332819696d0f8845d7ab73b72ceac54a8a41f2d11498d448aa",
      "idTokenA": "8",
      "idTokenB": "0",
      "key": "9",
      "name": "Mock Test Token-Defa,ult Defi token",
      "ownerAddress": "a914562bd6af924ee2c19e39a8dc962744e29e3d5fed87",
      "poolID": "9",
      "poolSharePercentage": "0",
      "reserveA": 12.73190551,
      "reserveA/reserveB": 0.16533442,
      "reserveB": 77.00698454,
      "reserveB/reserveA": 6.04834716,
      "rewardPct": 0,
      "status": true,
      "symbol": "MOCK-DFI",
      "tokenA": "MOCK",
      "tokenB": "DFI",
      "totalLiquidity": 31.31348449,
      "totalLiquidityInUSDT": "108.57984820",
      "tradeEnabled": true,
      "yearlyPoolReward": "0.00000000",
      "handleAddPoolLiquidity": "13a9ecc88c034e3a4482b0d13c0630466056d38152fafd3b694714ea11459efd"
    }
  ],
  "poolpair": [
    {
      "blockCommissionA": 4.2e-7,
      "blockCommissionB": 3e-8,
      "commission": 0.002,
      "creationHeight": 18145,
      "creationTx": "b2bb09ffe9f9b292f13d23bafa1225ef26d0b9906da7af194c5738b63839b235",
      "hash": "3",
      "idTokenA": "0",
      "idTokenB": "1",
      "name": "Default Defi token-",
      "ownerAddress": "76a914e5ed1bfa532e2358e964381a70e0df6d4dd0041988ac",
      "reserveA": 23500.25210905,
      "reserveA/reserveB": 2.77693382,
      "reserveB": 8462.66191877,
      "reserveB/reserveA": 0.3601094,
      "rewardPct": 0.85,
      "status": true,
      "symbol": "DFI-BTC",
      "tokenA": "DFI",
      "tokenB": "BTC",
      "totalLiquidity": 1726.8492621,
      "tradeEnabled": true
    }
  ],
  "addPoolLiquidityHash": "13a9ecc88c034e3a4482b0d13c0630466056d38152fafd3b694714ea11459efd",
  "removePoolLiquidityHash": "",
  "isErrorRemovingPoolLiquidity": "Error Message",
  "utxoDfi": 626.44518901,
  "isUtxoDfiError": "Error Message",
  "maxAccountDfi": 1.0,
  "MaxAccountDfiError": "Error Message",
  "saga": {
    "handleFetchPoolshares": [],
    "handleFetchTokenBalanceList": [],
    "handleAddPoolLiquidity": [],
    "handleRemovePoolLiquidity": [],
    "handleFetchUtxoDFI": 0,
    "handleFetchTokenDFI": 0
  },
  "poolshare": [
    {
      "%": 0,
      "amount": 0.58796283,
      "apy": "28.53",
      "blockCommissionA": 3.1e-7,
      "blockCommissionB": 5e-8,
      "commission": 0.002,
      "creationHeight": 18145,
      "creationTx": "b2bb09ffe9f9b292f13d23bafa1225ef26d0b9906da7af194c5738b63839b235",
      "hash": "3",
      "idTokenA": "0",
      "idTokenB": "1",
      "key": "3",
      "name": "Default Defi token-",
      "owner": "739nUiaW9SemqbaTvPr3dCRd4XyKD6GccD",
      "ownerAddress": "76a914e5ed1bfa532e2358e964381a70e0df6d4dd0041988ac",
      "poolID": "3",
      "poolSharePercentage": 6.977844749999999,
      "reserveA": 23502.68054771,
      "reserveA/reserveB": 2.80035108,
      "reserveB": 8392.76212895,
      "reserveB/reserveA": 0.35709808,
      "rewardPct": 0.85,
      "status": true,
      "symbol": "DFI-BTC",
      "tokenA": "DFI",
      "tokenB": "BTC",
      "totalLiquidity": 1719.79162218,
      "totalLiquidityInUSDT": "311997952.36457992",
      "tradeEnabled": true,
      "yearlyPoolReward": "109768932.00000000"
    },
    {
      "%": 0,
      "amount": 0.10033982,
      "apy": "253.32",
      "blockCommissionA": 2.2e-7,
      "blockCommissionB": 4e-8,
      "commission": 0.002,
      "creationHeight": 18145,
      "creationTx": "045c74ee64e2afce750d2dae5958a01023a0dd30105cff211b6fb715a6aa915d",
      "hash": "4",
      "idTokenA": "0",
      "idTokenB": "2",
      "key": "4",
      "name": "Default Defi token-",
      "owner": "74zqGjPVB5djKz9ZrP2SqA9wpsGBoj3YBL",
      "ownerAddress": "76a914e5ed1bfa532e2358e964381a70e0df6d4dd0041988ac",
      "poolID": "4",
      "poolSharePercentage": 11.32817626,
      "reserveA": 246182.6376474,
      "reserveA/reserveB": 100.63736612,
      "reserveB": 2446.23490384,
      "reserveB/reserveA": 0.00993666,
      "rewardPct": 0.1,
      "status": true,
      "symbol": "DFI-ETH",
      "tokenA": "DFI",
      "tokenB": "ETH",
      "totalLiquidity": 23422.61127153,
      "totalLiquidityInUSDT": "4133452.06501293",
      "tradeEnabled": true,
      "yearlyPoolReward": "12913992.00000000"
    },
    {
      "%": 0,
      "amount": 0.4131457,
      "apy": "918.46",
      "blockCommissionA": 1.1e-7,
      "blockCommissionB": 1e-8,
      "commission": 0.002,
      "creationHeight": 29584,
      "creationTx": "72fa270eddf0221113f5e29d0e9568f89e87c433c91aad50b1215f0aad60e7c3",
      "hash": "6",
      "idTokenA": "0",
      "idTokenB": "5",
      "key": "6",
      "name": "Default Defi token-",
      "owner": "76d85h2evzXwuePsHPhSGJSPTK1kA5n41m",
      "ownerAddress": "76a914e5ed1bfa532e2358e964381a70e0df6d4dd0041988ac",
      "poolID": "6",
      "poolSharePercentage": 0.20037192999999998,
      "reserveA": 193995.87607228,
      "reserveA/reserveB": 4.80089935,
      "reserveB": 40408.23643223,
      "reserveB/reserveA": 0.2082943,
      "rewardPct": 0.05,
      "status": true,
      "symbol": "DFI-USDT",
      "tokenA": "DFI",
      "tokenB": "USDT",
      "totalLiquidity": 79409.57427086,
      "totalLiquidityInUSDT": "570016.97810955",
      "tradeEnabled": true,
      "yearlyPoolReward": "6456996.00000000"
    },
    {
      "%": 78,
      "amount": 24.59449959,
      "apy": 0,
      "blockCommissionA": 0,
      "blockCommissionB": 2e-7,
      "commission": 0.002,
      "creationHeight": 87550,
      "creationTx": "c38593974982ce332819696d0f8845d7ab73b72ceac54a8a41f2d11498d448aa",
      "hash": "9",
      "idTokenA": "8",
      "idTokenB": "0",
      "key": "9",
      "name": "Mock Test Token-Default Defi token",
      "owner": "73dhtpEHb7y6HR75b6GRCghwNeTDpXacAT",
      "ownerAddress": "a914562bd6af924ee2c19e39a8dc962744e29e3d5fed87",
      "poolID": "9",
      "poolSharePercentage": "78.54283862",
      "reserveA": 12.73190551,
      "reserveA/reserveB": 0.16533442,
      "reserveB": 77.00698454,
      "reserveB/reserveA": 6.04834716,
      "rewardPct": 0,
      "status": true,
      "symbol": "MOCK-DFI",
      "tokenA": "MOCK",
      "tokenB": "DFI",
      "totalLiquidity": 31.31348449,
      "totalLiquidityInUSDT": "NaN",
      "tradeEnabled": true,
      "yearlyPoolReward": "0.00000000"
    }
  ]
}
