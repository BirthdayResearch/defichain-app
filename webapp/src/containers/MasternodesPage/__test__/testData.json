{
  "fetchMasternodesSuccess": [
    {
      "hash": "2a7ddddb0056bd814a580ec3a0420d6316a198e6ca60267f1d04cd6abe2ba31e",
      "ownerAuthAddress": "7BQZ67KKYWSmVRukgv57m4HorjbGh7NWrQ",
      "operatorAuthAddress": "7GULFtS6LuJfJEikByKKg8psscg84jnfHs",
      "creationHeight": 0,
      "resignHeight": -1,
      "resignTx": "0000000000000000000000000000000000000000000000000000000000000000",
      "banHeight": -1,
      "banTx": "0000000000000000000000000000000000000000000000000000000000000000",
      "state": "ENABLED",
      "mintedBlocks": 12008
    }
  ],
  "saga": {
    "handelCreateMasterNodes": {
      "masternodeOperator": "test_masternodeOperator",
      "masternodeOwner": "test_masternodeOwner",
      "masterNodeHash": "test_masterNodeHash"
    },
    "handleResignMasterNode": {},
    "getAddressInfo": {
      "address": "8Pf3bNJwTth7xW8RmK6qHd23PLnw14fhju",
      "scriptPubKey": "76a9145e07ce5fce8404d06c600a3f8abxxxxxxxxxxxxxxxx",
      "ismine": true,
      "solvable": true,
      "desc": "pkh(xxxxxxxxxxxxxxxxxxxx)#mdew0l3m",
      "iswatchonly": false,
      "isscript": false,
      "iswitness": false,
      "pubkey": "027dd7f5cc0cf5e20d2c17a7085dd75afca0426ebd95cxxxxxxxxxxxx",
      "iscompressed": true,
      "label": "",
      "ischange": false,
      "timestamp": 1,
      "labels": [
        {
          "name": "",
          "purpose": "receive"
        }
      ]
    }
  }
}
