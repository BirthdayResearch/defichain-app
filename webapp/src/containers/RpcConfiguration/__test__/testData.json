{
  "isBlockchainStarted": {
    "result": {
      "chain": "regtest",
      "blocks": 0,
      "headers": 0,
      "bestblockhash": "0f9188f13cb7b2c71f2a335e3a4fc328bf5beb436012afca590b1a11466e2206",
      "difficulty": 4.656542373906925e-10,
      "mediantime": 1296688602,
      "verificationprogress": 1,
      "initialblockdownload": true,
      "chainwork": "0000000000000000000000000000000000000000000000000000000000000002",
      "size_on_disk": 293,
      "pruned": false,
      "softforks": {
        "bip34": { "type": "buried", "active": false, "height": 500 },
        "bip66": { "type": "buried", "active": false, "height": 1251 },
        "bip65": { "type": "buried", "active": false, "height": 1351 },
        "csv": { "type": "buried", "active": false, "height": 432 },
        "segwit": { "type": "buried", "active": true, "height": 0 },
        "testdummy": {
          "type": "bip9",
          "bip9": {
            "status": "defined",
            "start_time": 0,
            "timeout": 9223372036854776000,
            "since": 0
          },
          "active": false
        }
      },
      "warnings": "This is a pre-release test build - use at your own risk - do not use for mining or merchant applications"
    },
    "error": null,
    "id": "curltest"
  }
}
