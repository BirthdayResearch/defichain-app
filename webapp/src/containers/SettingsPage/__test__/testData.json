{
  "settingsRowDropDown": {
    "languages": [
      {
        "label": "containers.settings.english",
        "value": "en"
      },
      {
        "label": "containers.settings.german",
        "value": "de"
      },
      {
        "label": "containers.settings.french",
        "value": "fr"
      }
    ],
    "amountUnits": [
      {
        "label": "DFI",
        "value": "DFI"
      },
      {
        "label": "fi",
        "value": "fi"
      }
    ],
    "displayModes": [
      {
        "label": "containers.settings.sameAsSystem",
        "value": "SAME_AS_SYSTEM"
      },
      {
        "label": "containers.settings.light",
        "value": "LIGHT"
      },
      {
        "label": "containers.settings.dark",
        "value": "DARK"
      }
    ],
    "language": "en-GB",
    "unit": "DFI",
    "displayMode": "SAME_AS_SYSTEM",
    "launchAtLogin": false,
    "minimizedAtLaunch": false
  },
  "settingTab": {
    "launchAtLogin": false,
    "minimizedAtLaunch": false,
    "language": "en-GB",
    "unit": "DFI",
    "displayMode": "SAME_AS_SYSTEM"
  },
  "settingTabDisplay": {
    "language": "en-GB",
    "unit": "DFI",
    "displayMode": "SAME_AS_SYSTEM"
  },
  "serviceTestData": {
    "initialData": {
      "blockStorage": "",
      "databaseCache": "",
      "displayMode": "SAME_AS_SYSTEM",
      "feeRate": 0.1,
      "language": "en",
      "launchAtLogin": false,
      "maximumAmount": 10000,
      "maximumCount": 1000,
      "minimizedAtLaunch": false,
      "pruneBlockStorage": false,
      "scriptVerificationThreads": 0,
      "unit": "DFI"
    }
  },
  "sagaTestData": {
    "getSettingsOptions": {
      "languages": [
        {
          "label": "containers.settings.english",
          "value": "en"
        },
        {
          "label": "containers.settings.german",
          "value": "de"
        },
        {
          "label": "containers.settings.french",
          "value": "fr"
        },
        {
          "label": "containers.settings.chinese",
          "value": "zhs"
        },
        {
          "label": "containers.settings.chinese_traditional",
          "value": "zht"
        },
        {
          "label": "containers.settings.dutch",
          "value": "nl"
        },
        {
          "label": "containers.settings.russian",
          "value": "ru"
        }
      ],
      "amountUnits": [
        {
          "label": "DFI",
          "value": "DFI"
        },
        {
          "label": "fi",
          "value": "fi"
        }
      ],
      "displayModes": [
        {
          "label": "containers.settings.sameAsSystem",
          "value": "SAME_AS_SYSTEM"
        },
        {
          "label": "containers.settings.light",
          "value": "LIGHT"
        },
        {
          "label": "containers.settings.dark",
          "value": "DARK"
        }
      ],
      "networkOptions": [
        {
          "label": "containers.settings.mainnet",
          "value": "Mainnet"
        },
        {
          "label": "containers.settings.testnet",
          "value": "Testnet"
        }
      ]
    },
    "updateSettings": {
      "language": "",
      "unit": "",
      "displayMode": "",
      "launchAtLogin": "",
      "minimizedAtLaunch": "",
      "pruneBlockStorage": "",
      "scriptVerificationThreads": "",
      "blockStorage": "",
      "databaseCache": ""
    }
  }
}
