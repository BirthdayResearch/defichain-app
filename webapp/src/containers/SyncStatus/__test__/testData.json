{
  "synStatus": {
    "latestBlock": 139114,
    "latestSyncedBlock": 139114,
    "syncedPercentage": "100.00"
  },
  "syncingError": "Error Message",
  "isLoading": false,
  "peers": 77
}
