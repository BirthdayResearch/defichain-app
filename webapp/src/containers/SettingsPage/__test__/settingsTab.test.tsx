import React from 'react';
import SettingsTabComponent from '../components/SettingsTab';
import Enzyme from '../../../utils/testUtils/enzyme';
import { Provider } from 'react-redux';
import store from '../../../app/rootStore';
import { settingTab } from './testData.json';

describe('SettingsTabComponent', () => {
  it('should check for snapshot', () => {
    const props = {
      ...settingTab,
      handleDropDowns: () => {},
      handleToggles: () => {},
    };
    const wrapper = Enzyme.mount(
      <Provider store={store}>
        <SettingsTabComponent {...props} />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
