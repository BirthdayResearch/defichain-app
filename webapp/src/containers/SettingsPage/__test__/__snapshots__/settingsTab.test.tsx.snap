// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsTabComponent should check for snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(SettingsTab)
    displayMode="SAME_AS_SYSTEM"
    handleDropDowns={[Function]}
    handleToggles={[Function]}
    language="en-GB"
    launchAtLogin={false}
    minimizedAtLaunch={false}
    unit="DFI"
  >
    <SettingsTab
      amountUnits={Array []}
      dispatch={[Function]}
      displayMode="SAME_AS_SYSTEM"
      displayModes={Array []}
      handleDropDowns={[Function]}
      handleToggles={[Function]}
      language="en-GB"
      languages={Array []}
      launchAtLogin={false}
      minimizedAtLaunch={false}
      unit="DFI"
    >
      <TabPane
        tabId="display"
        tag="div"
      >
        <div
          className="tab-pane"
        >
          <section>
            <Form
              tag="form"
            >
              <form
                className=""
              >
                <Row
                  className="mb-5"
                  tag="div"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ]
                  }
                >
                  <div
                    className="mb-5 row"
                  >
                    <Col
                      md="4"
                      tag="div"
                      widths={
                        Array [
                          "xs",
                          "sm",
                          "md",
                          "lg",
                          "xl",
                        ]
                      }
                    >
                      <div
                        className="col-md-4"
                      >
                        Launch options
                      </div>
                    </Col>
                    <Col
                      md="8"
                      tag="div"
                      widths={
                        Array [
                          "xs",
                          "sm",
                          "md",
                          "lg",
                          "xl",
                        ]
                      }
                    >
                      <div
                        className="col-md-8"
                      >
                        <Connect(SettingsRowToggle)
                          field={false}
                          fieldName="launchAtLogin"
                          handleToggles={[Function]}
                          label="launchAtLogin"
                        >
                          <SettingsRowToggle
                            dispatch={[Function]}
                            field={false}
                            fieldName="launchAtLogin"
                            handleToggles={[Function]}
                            label="launchAtLogin"
                            locale="en"
                          >
                            <FormGroup
                              className=""
                              tag="div"
                            >
                              <div
                                className="form-group"
                              >
                                <FormGroup
                                  check={true}
                                  tag="div"
                                >
                                  <div
                                    className="form-check"
                                  >
                                    <Label
                                      check={true}
                                      className="switch"
                                      tag="label"
                                      widths={
                                        Array [
                                          "xs",
                                          "sm",
                                          "md",
                                          "lg",
                                          "xl",
                                        ]
                                      }
                                    >
                                      <label
                                        className="switch form-check-label"
                                      >
                                        <Input
                                          checked={false}
                                          onChange={[Function]}
                                          type="checkbox"
                                        >
                                          <input
                                            checked={false}
                                            className="form-check-input"
                                            onChange={[Function]}
                                            type="checkbox"
                                          />
                                        </Input>
                                         
                                        Launch at login
                                      </label>
                                    </Label>
                                  </div>
                                </FormGroup>
                              </div>
                            </FormGroup>
                          </SettingsRowToggle>
                        </Connect(SettingsRowToggle)>
                        <Connect(SettingsRowToggle)
                          field={false}
                          fieldName="minimizedAtLaunch"
                          handleToggles={[Function]}
                          hideMinimized={true}
                          label="minimizedAtLaunch"
                        >
                          <SettingsRowToggle
                            dispatch={[Function]}
                            field={false}
                            fieldName="minimizedAtLaunch"
                            handleToggles={[Function]}
                            hideMinimized={true}
                            label="minimizedAtLaunch"
                            locale="en"
                          >
                            <FormGroup
                              className="d-none"
                              tag="div"
                            >
                              <div
                                className="d-none form-group"
                              >
                                <FormGroup
                                  check={true}
                                  tag="div"
                                >
                                  <div
                                    className="form-check"
                                  >
                                    <Label
                                      check={true}
                                      className="switch"
                                      tag="label"
                                      widths={
                                        Array [
                                          "xs",
                                          "sm",
                                          "md",
                                          "lg",
                                          "xl",
                                        ]
                                      }
                                    >
                                      <label
                                        className="switch form-check-label"
                                      >
                                        <Input
                                          checked={false}
                                          onChange={[Function]}
                                          type="checkbox"
                                        >
                                          <input
                                            checked={false}
                                            className="form-check-input"
                                            onChange={[Function]}
                                            type="checkbox"
                                          />
                                        </Input>
                                         
                                        Minimized at launch
                                      </label>
                                    </Label>
                                  </div>
                                </FormGroup>
                              </div>
                            </FormGroup>
                          </SettingsRowToggle>
                        </Connect(SettingsRowToggle)>
                      </div>
                    </Col>
                  </div>
                </Row>
                <Connect(SettingsRowDropDown)
                  data={Array []}
                  field="en-GB"
                  fieldName="language"
                  handleDropDowns={[Function]}
                  label="containers.settings.appLanguage"
                >
                  <SettingsRowDropDown
                    data={Array []}
                    dispatch={[Function]}
                    field="en-GB"
                    fieldName="language"
                    handleDropDowns={[Function]}
                    label="containers.settings.appLanguage"
                    locale="en"
                  >
                    <FormGroup
                      className="form-row align-items-center"
                      tag="div"
                    >
                      <div
                        className="form-row align-items-center form-group"
                      >
                        <Col
                          md="4"
                          tag="div"
                          widths={
                            Array [
                              "xs",
                              "sm",
                              "md",
                              "lg",
                              "xl",
                            ]
                          }
                        >
                          <div
                            className="col-md-4"
                          >
                            App language
                          </div>
                        </Col>
                        <Col
                          md="8"
                          tag="div"
                          widths={
                            Array [
                              "xs",
                              "sm",
                              "md",
                              "lg",
                              "xl",
                            ]
                          }
                        >
                          <div
                            className="col-md-8"
                          >
                            <UncontrolledDropdown>
                              <Dropdown
                                a11y={true}
                                active={false}
                                addonType={false}
                                direction="down"
                                inNavbar={false}
                                isOpen={false}
                                nav={false}
                                setActiveFromChild={false}
                                toggle={[Function]}
                              >
                                <Manager>
                                  <div
                                    className="dropdown"
                                    onKeyDown={[Function]}
                                  >
                                    <DropdownToggle
                                      aria-haspopup={true}
                                      caret={true}
                                      color="outline-secondary"
                                      disabled={false}
                                    >
                                      <Reference>
                                        <InnerReference
                                          setReferenceNode={[Function]}
                                        >
                                          <Button
                                            aria-expanded={false}
                                            aria-haspopup={true}
                                            className="dropdown-toggle"
                                            color="outline-secondary"
                                            disabled={false}
                                            innerRef={[Function]}
                                            onClick={[Function]}
                                            tag="button"
                                          >
                                            <button
                                              aria-expanded={false}
                                              aria-haspopup={true}
                                              aria-label={null}
                                              className="dropdown-toggle btn btn-outline-secondary"
                                              disabled={false}
                                              onClick={[Function]}
                                              type="button"
                                            />
                                          </Button>
                                        </InnerReference>
                                      </Reference>
                                    </DropdownToggle>
                                    <DropdownMenu
                                      flip={true}
                                      tag="div"
                                    >
                                      <div
                                        aria-hidden={true}
                                        className="dropdown-menu"
                                        role="menu"
                                        tabIndex="-1"
                                      />
                                    </DropdownMenu>
                                  </div>
                                </Manager>
                              </Dropdown>
                            </UncontrolledDropdown>
                          </div>
                        </Col>
                      </div>
                    </FormGroup>
                  </SettingsRowDropDown>
                </Connect(SettingsRowDropDown)>
              </form>
            </Form>
          </section>
        </div>
      </TabPane>
    </SettingsTab>
  </Connect(SettingsTab)>
</Provider>
`;
